dowhy/causal_estimator.py

```diff
@@ -679,6 +685,91 @@ def signif_results_tostr(self, signif_results):
+        * Regression Discontinuity: "iv.regression_discontinuity"
+
+    In addition, you can directly call any of the EconML estimation methods. The convention is "backdoor.econml.path-to-estimator-class". For example, for the double machine learning estimator ("DML" class) that is located inside "dml" module of EconML, you can use the method name, "backdoor.econml.dml.DML". CausalML estimators can also be called. See `this demo notebook <https://py-why.github.io/dowhy/example_notebooks/dowhy-conditional-treatment-effects.html>`_.
+
+
+    :param identified_estimand: a probability expression
+        that represents the effect to be estimated. Output of
+        CausalModel.identify_effect method
+    :param method_name: name of the estimation method to be used.
+    :param control_value: Value of the treatment in the control group, for effect estimation.  If treatment is multi-variate, this can be a list.
+    :param treatment_value: Value of the treatment in the treated group, for effect estimation. If treatment is multi-variate, this can be a list.
+    :param test_significance: Binary flag on whether to additionally do a statistical signficance test for the estimate.
+    :param evaluate_effect_strength: (Experimental) Binary flag on whether to estimate the relative strength of the treatment's effect. This measure can be used to compare different treatments for the same outcome (by running this method with different treatments sequentially).
+    :param confidence_intervals: (Experimental) Binary flag indicating whether confidence intervals should be computed.
+    :param target_units: (Experimental) The units for which the treatment effect should be estimated. This can be of three types. (1) a string for common specifications of target units (namely, "ate", "att" and "atc"), (2) a lambda function that can be used as an index for the data (pandas DataFrame), or (3) a new DataFrame that contains values of the effect_modifiers and effect will be estimated only for this new data.
+    :param effect_modifiers: Names of effect modifier variables can be (optionally) specified here too, since they do not affect identification. If None, the effect_modifiers from the CausalModel are used.
+    :param fit_estimator: Boolean flag on whether to fit the estimator.
+        Setting it to False is useful to estimate the effect on new data using a previously fitted estimator.
+    :param method_params: Dictionary containing any method-specific parameters. These are passed directly to the estimating method. See the docs for each estimation method for allowed method-specific params.
+    :returns: An instance of the CausalEstimate class, containing the causal effect estimate
+        and other method-dependent information
+
+    """
+    treatment = parse_state(treatment)
+    outcome = parse_state(outcome)
+    causal_estimator_class = method.__class__
+
+    identified_estimand.set_identifier_method(identifier_name)
+
+    if fit_estimator:
```

Comment:
since we are now passing the method object, fit_estimator makes more sense for the next line, the estimate_effect method of the Estimator. 

Here we can simply do the two if-else checks and then sequentially do "method.update_input". Just the "if fit_estimator" line can be removed.